x/snapshot/keeper/msg_server.go

```diff
@@ -83,5 +87,19 @@ func (s msgServer) DeactivateProxy(c context.Context, req *types.DeactivateProxy
 		})
 
 	s.Keeper.Logger(ctx).Info(fmt.Sprintf("validator %s has de-activated proxy %s", req.Sender.String(), proxy.String()))
+
+	// remove validator as chain maintainer since it can no longer vote
+	for _, chain := range s.nexus.GetChains(ctx) {
+		if !s.nexus.IsChainMaintainer(ctx, chain, req.Sender) {
+			continue
+		}
+
+		s.Logger(ctx).Info(fmt.Sprintf("validator %s deregistered maintainer for chain %s", req.Sender.String(), chain.Name))
+
+		if err := s.nexus.RemoveChainMaintainer(ctx, chain, req.Sender); err != nil {
```

Comment:
This [comment](https://github.com/axelarnetwork/axelar-core/pull/1102/files#r770089769) is also relevant here.